src/interaction/keyframenavigator.cpp

```diff
@@ -107,22 +124,62 @@ void KeyframeNavigator::updateCamera(Camera& camera) {
     nextKeyframeCameraPosition += nextFocusNode->worldPosition();
 
     // Linear interpolation
-    camera.setPositionVec3(
+    t = std::max(0.0, std::min(1.0, t));
+    camera->setPositionVec3(
         prevKeyframeCameraPosition * (1 - t) + nextKeyframeCameraPosition * t
     );
-    camera.setRotation(
+    camera->setRotation(
         glm::slerp(prevKeyframeCameraRotation, nextKeyframeCameraRotation, t)
     );
-
+    
     // We want to affect view scaling, such that we achieve
     // logarithmic interpolation of distance to an imagined focus node.
     // To do this, we interpolate the scale reciprocal logarithmically.
-    const float prevInvScaleExp = glm::log(1.f / prevPose.scale);
-    const float nextInvScaleExp = glm::log(1.f / nextPose.scale);
-    const float interpolatedInvScaleExp = static_cast<float>(
-        prevInvScaleExp * (1 - t) + nextInvScaleExp * t
-    );
-    camera.setScaling(1.f / glm::exp(interpolatedInvScaleExp));
+    if (!ignoreFutureKeyframes) {
+        const float prevInvScaleExp = glm::log(1.f / prevPose.scale);
+        const float nextInvScaleExp = glm::log(1.f / nextPose.scale);
+        const float interpolatedInvScaleExp = static_cast<float>(
+            prevInvScaleExp * (1 - t) + nextInvScaleExp * t
+            );
+        camera->setScaling(1.f / glm::exp(interpolatedInvScaleExp));
+    }
+
+#ifdef INTERPOLATION_DEBUG_PRINT
+    LINFO(fmt::format("Cam pos prev={}, next={}", std::to_string(prevKeyframeCameraPosition),
+        std::to_string(nextKeyframeCameraPosition)));
+    LINFO(fmt::format("Cam rot prev={} {} {} {}  next={} {} {} {}", prevKeyframeCameraRotation.x,
+        prevKeyframeCameraRotation.y, prevKeyframeCameraRotation.z, prevKeyframeCameraRotation.w,
+        nextKeyframeCameraRotation.x, nextKeyframeCameraRotation.y, nextKeyframeCameraRotation.z,
+        nextKeyframeCameraRotation.w));
+    LINFO(fmt::format("Cam interp = {}", t));
+    LINFO(fmt::format("camera {} {} {} {} {} {}", global::windowDelegate.applicationTime(),
+        global::windowDelegate.applicationTime() - _timestampPlaybackStarted_application,
+        global::timeManager.time().j2000Seconds(), interpolatedCamera.x,
+        interpolatedCamera.y, interpolatedCamera.z));
+    //Following is for direct print to save & compare camera positions against recorded file
+    printf("camera %8.4f %8.4f %13.3f %16.7f %16.7f %16.7f\n", global::windowDelegate.applicationTime(),
+        global::windowDelegate.applicationTime() - _timestampPlaybackStarted_application,
+        global::timeManager.time().j2000Seconds(), interpolatedCamera.x,
+        interpolatedCamera.y, interpolatedCamera.z);
+
+#endif
+
+
+    return true;
+}
+
+double KeyframeNavigator::currentTime() const {
+    if( _timeframeMode == KeyframeTimeRef::relative_recordedStart )
```

Comment:
Braces around single line statements { ... }
Space after `if` and before `(`

Reply:
Changed